core/src/main/java/com/hamdan/forzenbook/core/UtilFunctions.kt

```diff
@@ -46,20 +44,23 @@ fun getBitmapFromUri(
         .into(object : SimpleTarget<Bitmap>() {
             override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
                 // Save the downloaded bitmap to a temporary file
-                val tempFile: File
+                var tempFile: File? = null
```

Comment:
I don't see the reason to make this null here and then pass null to your callback and handle an error there. You should bring in two lambdas here, one for saved and one for error so the code can be cleanly seaprated